packages/utils/src/state/state-object.ts

```diff
@@ -0,0 +1,31 @@
+/* Internal Imports */
+import { abi } from '../utils'
+
+const STATE_OBJECT_ABI_TYPES = ['address', 'bytes']
+
+export interface StateObjectData {
+  predicate: string
+  parameters: string
+}
+
+/**
+ * Class that represents a simple state object.
+ * State objects are the fundamental building
+ * blocks of our plasma chain design.
+ */
+export class StateObject {
+  public predicate: string
+  public parameters: string
+
+  constructor(args: StateObjectData) {
+    this.predicate = args.predicate
+    this.parameters = args.parameters
+  }
```

Comment:
Curious why break it into StateObjectData & StateObject instead of just passing the args directly into the constructor. This is clearly very intentional so I'm just curious what I'm missing

Reply:
Personal preference. I like the following construction:

```
const StateObject = new StateObject({
  predicate: ‘0x...’,
  parameters: ...
})
```

Main advantage is that it’s clear what you’re passing into the constructor w/o having to look at the constructor definition.